lib/improver/spotdata/extract_data.py

```diff
@@ -227,44 +288,65 @@ def make_cube(self, cube, data, sites):
             at the spotdata sites.
 
         """
-        # Ensure time is a dimension coordinate.
+
+        # Ensure time is a dimension coordinate and convert to seconds.
         if 'time' not in cube.dim_coords:
             cube = iris.util.new_axis(cube, 'time')
+        cube.coord('time').convert_units('seconds since 1970-01-01 00:00:00')
 
+        # Replicate all non spatial dimension coodinates.
         n_non_spatial_dimcoords = len(cube.dim_coords) - 2
         non_spatial_dimcoords = cube.dim_coords[0:n_non_spatial_dimcoords]
-        cube.coord('time').convert_units('hours since 1970-01-01 00:00:00')
-
-        latitudes = [float(site['latitude']) for site in sites.itervalues()]
-        longitudes = [float(site['longitude']) for site in sites.itervalues()]
-        altitudes = [np.nan_to_num(site['altitude'])
-                     for site in sites.itervalues()]
-        utc_offsets = [float(site['utc_offset'])
-                       for site in sites.itervalues()]
-        wmo_sites = [site['wmo_site'] for site in sites.itervalues()]
-
-        indices, latitude, longitude, altitude, utc_offset, wmo_site = (
-            self._build_coordinates(
-                latitudes, longitudes, altitudes, utc_offsets, wmo_sites))
-
         dim_coords = [coord for coord in non_spatial_dimcoords]
+
+        # Add an index coordinate as a dimension coordinate.
+        indices = self._build_coordinate(np.arange(len(sites)), 'index',
+                                         data_type=int)
         dim_coords.append(indices)
+
+        # Include the forecast_reference_time from the source cube.
+        try:
+            forecast_ref_time = cube.coord('forecast_reference_time')
+        except CoordinateNotFoundError:
+            raise CoordinateNotFoundError(
+                'No forecast reference time found on source cube.')
+
+        forecast_ref_time.convert_units('seconds since 1970-01-01 00:00:00')
+        forecast_periods = cube.coord('time').points - forecast_ref_time.points
+        forecast_period = self._build_coordinate(
+            forecast_periods, 'forecast_period', units='seconds')
+
+        # Build the auxiliary coordinates.
+        crds = self._aux_coords_to_make()
+        aux_crds = []
+        for key, kwargs in zip(crds.keys(), crds.itervalues()):
+            aux_data = np.array([entry[key] for entry in sites.itervalues()])
+            crd = self._build_coordinate(aux_data, key, **kwargs)
+            aux_crds.append(crd)
+
+        # Construct zipped lists of coordinates and indices.
         n_dim_coords = len(dim_coords)
         dim_coords = zip(dim_coords, range(n_dim_coords))
-        aux_coords = zip([latitude, longitude, altitude, utc_offset, wmo_site],
-                         [n_dim_coords-1]*5)
+        aux_coords = zip(aux_crds, [n_dim_coords-1]*len(aux_crds))
+
```

Comment:
e.g. Scalar coordinates height: 1.5m for air_temperature cube (or would this get pick up by metadata?)